logstash-core/src/main/java/org/logstash/common/clients/HttpClient.java

```diff
@@ -0,0 +1,701 @@
+                    } else {
+                        sslSocketFactory = new SSLConnectionSocketFactory(sslContext, SSLConnectionSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER);
+                    }
+
+                    httpClientBuilder.setSSLSocketFactory(sslSocketFactory);
+                } catch (Exception e) {
+                    throw new OptionsBuilderException("Unable to configure HTTP client with SSL/TLS", e);
+                }
+            }
+
+            if (usesBasicAuth()) {
+                String credentials = this.basicAuthUsername + ":" + this.basicAuthPassword;
+                String encodedCredentials = new String(Base64.getEncoder().encode(credentials.getBytes(StandardCharsets.UTF_8)), StandardCharsets.UTF_8);
+
+                List<Header> headerList = Collections.singletonList(new BasicHeader("Authorization", "Basic " + encodedCredentials));
+                httpClientBuilder.setDefaultHeaders(headerList);
+            }
+
+            return new HttpClient(httpClientBuilder.build(), baseUrl);
+        }
+
+        private boolean usesBasicAuth() {
+            return (this.basicAuthUsername != null) && (this.basicAuthPassword != null);
+        }
+
+        private boolean usesSsl(URL baseUrl) {
+            return baseUrl.getProtocol().equals("https");
+        }
+
+        private static X509Certificate getCertificate(String certificateFilePath) throws OptionsBuilderException {
+            FileInputStream fis;
+            try {
+                fis = new FileInputStream(certificateFilePath);
+            } catch (FileNotFoundException e) {
+                throw new OptionsBuilderException("Could not read certificate at " + certificateFilePath, e);
+            }
+
+            try {
+                CertificateFactory cf = CertificateFactory.getInstance("X.509");
+                return (X509Certificate) cf.generateCertificate(fis);
+            } catch (Exception e) {
+                throw new OptionsBuilderException("Could not generate certificate from " + certificateFilePath, e);
+
+            } finally {
+                try {
+                    fis.close();
+                } catch (IOException e) {
+                    throw new OptionsBuilderException("Could not close certificate file at " + certificateFilePath + " after reading it", e);
+                }
+            }
+        }
+
+        private static RSAPrivateKey getPrivateKey(String privateKeyPath) throws OptionsBuilderException {
+            String privateKeyFileContents;
+            try {
+                privateKeyFileContents = new String(Files.readAllBytes(Paths.get(privateKeyPath)));
+            } catch (IOException e) {
+                throw new OptionsBuilderException("Could not read private key file at " + privateKeyPath, e);
+            }
+
+            final Pattern KEY_EXTRACTION_REGEXP = Pattern.compile(".*-----BEGIN (\\S+ )?PRIVATE KEY-----\n(.*)-----END (\\S+ )?PRIVATE KEY.*$", Pattern.DOTALL);
```

Comment:
I think you can just move this `Pattern` to a `private static final` constant, makes the code a little more idiomatic :)